filter/challenges-08.test.js

```diff
@@ -0,0 +1,280 @@
+        name: 'special-attack',
+      },
+      effort: 9,
+      baseStat: 65,
+    },
+  ],
+  name: 'snorlax',
+  weight: 4600,
+};
+
+const getBaseStatGreaterThan = (input, minBaseStat) => {
+  return input.filter( obj => obj.baseStat > minBaseStat)
+  };
+// ------------------------------------------------------------------------------------------------
+// CHALLENGE 5
+//
+// Write a function named getStatName that is an extension of your getBaseStatGreaterThan function
+// from challenge 4. For this function, extend your solution from challenge 4 to only return
+// the name of the stat, rather than the entire stat object.
+//
+// For example, getStatName(snorlaxData.stats, 50) will return ['special-defense', 'special-attack'].
+// ------------------------------------------------------------------------------------------------
+
+const getStatName = (input, minBaseStat) => {
+  return input.filter( obj => obj.baseStat > minBaseStat).map( thing => thing.stat.name);
+};
+
+// ------------------------------------------------------------------------------------------------
+// CHALLENGE 6
+//
+// Write a function named getCharactersWithoutChildren that, given the array of characters, below,
+// uses filter to return an array of all characters without children.
+// ------------------------------------------------------------------------------------------------
+
+const characters = [
+  {
+    name: 'Eddard',
+    spouse: 'Catelyn',
+    children: ['Robb', 'Sansa', 'Arya', 'Bran', 'Rickon'],
+    house: 'Stark',
+  },
+  {
+    name: 'Jon',
+    spouse: 'Lysa',
+    children: ['Robin'],
+    house: 'Arryn',
+  },
+  {
+    name: 'Cersei',
+    spouse: 'Robert',
+    children: ['Joffrey', 'Myrcella', 'Tommen'],
+    house: 'Lannister',
+  },
+  {
+    name: 'Daenarys',
+    spouse: 'Khal Drogo',
+    children: ['Drogon', 'Rhaegal', 'Viserion'],
+    house: 'Targaryen',
+  },
+  {
+    name: 'Mace',
+    spouse: 'Alerie',
+    children: ['Margaery', 'Loras'],
+    house: 'Tyrell',
+  },
+  {
+    name: 'Sansa',
+    spouse: 'Tyrion',
+    house: 'Stark',
+  },
+  {
+    name: 'Jon',
+    spouse: null,
+    house: 'Snow',
+  },
+];
+
+const getCharactersWithoutChildren = (input) => {
+  return input.filter( obj => !obj.children);
+};
+
+// ------------------------------------------------------------------------------------------------
+// CHALLENGE 7
+//
+// Write a function named evenOddNumericValues that, given an array as input, uses filter to
+// remove any non-numeric values, then uses map to generate a new array containing the string
+// 'even' or 'odd', depending on the original value.
+//
+// For example: evenOddNumericValues(['Gregor', 2, 4, 1]) returns ['even', 'even', 'odd'].
+// ------------------------------------------------------------------------------------------------
+
+const evenOddNumericValues = (input) => {
+  return input.filter( isNum => Number.isInteger(isNum)).map( num => (num%2===0)? 'even' : 'odd');
```

Comment:
That'll remove anything that is a number, but isn't an integer! It passes the tests, but I'd rather use `typeof` to avoid that problem.